tests/override_config_table/test_override_config_masic.py

```diff
@@ -0,0 +1,172 @@
+CONFIG_DB = "/etc/sonic/config_db.json"
+CONFIG_DB_BACKUP = "/etc/sonic/config_db.json_before_override"
+NON_USER_CONFIG_TABLES = ["FLEX_COUNTER_TABLE"]
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology('t2', 't1'),
+    pytest.mark.disable_loganalyzer,
+]
+
+
+@pytest.fixture(scope="module", autouse=True)
+def check_image_version(duthost):
+    """Skips this test if the SONiC image installed on DUT is older than 202111
+
+    Args:
+        duthost: DUT host object.
+
+    Returns:
+        None.
+    """
+    skip_release(duthost, ["201811", "201911", "202012", "202106", "202111"])
+
+
+@pytest.fixture(scope="module")
+def golden_config_exists_on_dut(duthost):
+    return file_exists_on_dut(duthost, GOLDEN_CONFIG)
+
+
+@pytest.fixture(scope="module")
+def setup_env(duthost, golden_config_exists_on_dut, tbinfo):
+    """
+    Setup/teardown
+    Args:
+        duthost: DUT.
+        golden_config_exists_on_dut: Check if golden config exists on DUT.
+    """
+    topo_type = tbinfo["topo"]["type"]
+    if topo_type in ["m0", "mx"]:
+        original_pfcwd_value = update_pfcwd_default_state(duthost, "/etc/sonic/init_cfg.json", "disable")
+    # Backup configDB
+    backup_config(duthost, CONFIG_DB, CONFIG_DB_BACKUP)
+    # Backup Golden Config if exists.
+    if golden_config_exists_on_dut:
+        backup_config(duthost, GOLDEN_CONFIG, GOLDEN_CONFIG_BACKUP)
+
+    # Reload test env with minigraph
+    config_reload(duthost, config_source="minigraph", safe_reload=True)
+    running_config = get_running_config(duthost)
+
+    yield running_config
+
+    if topo_type in ["m0", "mx"]:
+        update_pfcwd_default_state(duthost, "/etc/sonic/init_cfg.json", original_pfcwd_value)
+    # Restore configDB after test.
+    restore_config(duthost, CONFIG_DB, CONFIG_DB_BACKUP)
+    # Restore Golden Config after test, else cleanup test file.
+    if golden_config_exists_on_dut:
+        restore_config(duthost, GOLDEN_CONFIG, GOLDEN_CONFIG_BACKUP)
+    else:
+        duthost.file(path=GOLDEN_CONFIG, state='absent')
+
+    # Restore config before test
+    config_reload(duthost)
+
+
+def load_minigraph_with_golden_empty_input(duthost):
+    """Test Golden Config with empty input
+    """
+    initial_config = get_running_config(duthost)
+
+    empty_input = {}
+    reload_minigraph_with_golden_config(duthost, empty_input)
+
+    current_config = get_running_config(duthost)
+    for table in initial_config:
+        if table in NON_USER_CONFIG_TABLES:
+            continue
+        pytest_assert(
+            initial_config[table] == current_config[table],
+            "empty input compare fail! {}".format(table)
+        )
+
+
+def load_minigraph_with_golden_partial_config(duthost):
+    """Test Golden Config with partial config.
+
+    Here we assume all config contain SYSLOG_SERVER table
+    """
+    partial_config = {
+        "localhost": {},
+        "asic0": {},
+        "SYSLOG_SERVER": {
```

Comment:
Hi this is not aligned with the design. It should be the format in below:
```
partial_config = {
  "localhost": {
    "SYSLOG_SERVER": {
      "10.0.0.100": {}
    }
  },
  "asic0": {
      "SYSLOG_SERVER": {
        "10.0.0.100": {}
      }
  }
}
```
If I didn't remember it wrong, the syslog server should only be in localhost. So it may not be a good table to do masic override.